src/main/java/com/Elessar/app/server/MyServer.java

```diff
@@ -21,27 +26,40 @@
     private final String serverName;
     private final int port;
     private final MyDatabase db;
+    private final OperationsMonitor monitor;
+    private static Timer timer;
 
     public MyServer(String serverName, int port, MyDatabase db) {
         this.serverName = serverName;
         this.port = port;
         this.db = db;
+        this.monitor = new OperationsMonitor();
+        timer = new Timer();
     }
 
     public void run() {
         try {
             final HttpServer server = HttpServer.create(new InetSocketAddress(serverName, port), 0);
             server.createContext("/", new RootHandler());
             server.createContext("/echo", new EchoHandler());
-            server.createContext("/register", new RegisterHandler(db));
-            server.createContext("/logon", new LogOnHandler(db));
-            server.createContext("/logoff", new LogOffHandler(db));
+            server.createContext("/register", new RegisterHandler(db, monitor));
+            server.createContext("/logon", new LogOnHandler(db, monitor));
+            server.createContext("/logoff", new LogOffHandler(db, monitor));
             server.setExecutor(null);
             server.start();
             logger.info("Server started at port {}", port);
         } catch (IOException e) {
             logger.fatal("Caught exception during server startup: {}", e);
         }
+        timer.scheduleAtFixedRate(new TimerTask() {
+            @Override
+            public void run() {
+                for (String operation : monitor.getOperationSet()) {
+                    logger.printf(Level.INFO, "%s operation takes %.2f ms on average", operation, monitor.getAvgTime(operation));
```

Comment:
I recommend to log the metrics in key-value pairs or CSV to make it easy to parse when we want to analyze the metrics. For example:
name=db.user.insert,count=4,avg_time=123.4

And we probably want to disable logging unnecessary information such as log level and thread id. Those are configurable.